Data/EmulatedDevice.cs

```diff
@@ -221,14 +221,57 @@ public ICommand AddKeyCommand
             }
         }
 
+        [JsonIgnore]
+        public ICommand AddAllKeyCommand
+        {
+            get { return new RelayCommand(AddAllKey); }
+        }
+
+        private void AddAllKey(object obj)
+        {
+            string name;
+            while ((name = GetNextSugges()) != "")
+            {
+                AddKey(name);
+            }
+        }
+
         private void AddKey(object obj)
         {
+            string name;
+            name = obj == null ? GetNextSugges() : obj.ToString();
+
             EmulatedKeys.Add(
                 new EmulatedKey
                 {
-                    Name = ""
+                    Name = name
                 });
         }
+
+        /// <summary>
+        /// Search for a key name that is not in use.
+        /// </summary>
+        private string GetNextSugges()
+        {
+            foreach (string suggestions in SearchSuggestions)
```

Comment:
This function seems a bit odd, in particular the use of `IndexOf` when you already are looping.

I think using `Find` in conjunction with null coalescing can help:

```
if (EmulatedKeys.Count == 0)
{
  if (SearchSuggestions.Count == 0)
  {
  // No keys or suggestions, return empty
    return "";
  }

  // No keys, just return the first suggestion
  return SearchSuggestions[0];
}

var suggestion_found = SearchSuggestions.Find(suggestion => EmulatedKeys.Find(key => key.Name != suggestion) != null);

// return suggestion or empty
return suggestion_found ?? "";
```

Reply:
This is a bit strange for me...
especially because ObservableCollection doesn't have a Find function and we would have to convert them to lists first.
but i think i have found a clean [alternative](https://github.com/iwubcode/DolphinDynamicInputTextureCreator/pull/16/commits/91e6fc0908ed9f2b0550e0a3edce668f5c918291).

Reply:
Thanks for baring with me @Venomalia !  I thought we were dealing with lists, sorry.

Anyway, I think `goto` should be avoided in modern code-bases (except under extreme circumstances) so I'd probably prefer this:

```
foreach (string suggestion in SearchSuggestions)
{
	bool skip_suggestion = false;
	foreach (EmulatedKey key in EmulatedKeys)
	{
		if (key.Name == suggestion)
			skip_suggestion = true;
	}
	
	if (!skip_suggestion)
		return suggestion;
}
return "";
```

are you ok with that?

Reply:
my ideas are also sometimes unusual ¯\_(ツ)_/¯